third_party/mbedtls/native_its/src/file/its_file.c

```diff
@@ -0,0 +1,384 @@
+}
+
+/**
+ * Reads an 8-byte header from the given file:
+ * - 4 bytes for flags (psa_storage_create_flags_t)
+ * - 4 bytes for data length
+ *
+ * Returns 0 on success, -1 on error.
+ */
+static int readHeader(FILE *aFile, psa_storage_create_flags_t *aFlags, uint32_t *aDataLen)
+{
+    int result = 0;
+    uint32_t flagsTmp;
+    uint32_t lenTmp;
+
+    // Read flags
+    VerifyOrExit(fread(&flagsTmp, sizeof(flagsTmp), 1, aFile) == 1, result = -1);
+
+    // Read length
+    VerifyOrExit(fread(&lenTmp, sizeof(lenTmp), 1, aFile) == 1, result = -1);
+
+    *aFlags   = flagsTmp;
+    *aDataLen = lenTmp;
+
+exit:
+    return result;
+}
+
+/**
+ * Writes an 8-byte header to the given file:
+ * - 4 bytes for flags
+ * - 4 bytes for data length
+ *
+ * Returns 0 on success, -1 on error.
+ */
+static int writeHeader(FILE *aFile, psa_storage_create_flags_t aFlags, uint32_t aDataLen)
+{
+    int result = 0;
+
+    // Write flags
+    VerifyOrExit(fwrite(&aFlags, sizeof(aFlags), 1, aFile) == 1, result = -1);
+
+    // Write length
+    VerifyOrExit(fwrite(&aDataLen, sizeof(aDataLen), 1, aFile) == 1, result = -1);
+
+exit:
+    return result;
+}
+
+psa_status_t psa_its_set(psa_storage_uid_t           aUid,
+                         uint32_t                    aDataLength,
+                         const void                 *aData,
+                         psa_storage_create_flags_t  aCreateFlags)
+{
+    psa_status_t status = PSA_SUCCESS;
+    FILE        *file   = NULL;
+    char         path[ITS_FILE_PATH_MAX];
+
+    // Validate arguments
+    VerifyOrExit((aData != NULL && aDataLength > 0), status = PSA_ERROR_INVALID_ARGUMENT);
+    // Only NONE or WRITE_ONCE => no other flags supported
+    VerifyOrExit((aCreateFlags & ~(PSA_STORAGE_FLAG_WRITE_ONCE)) == 0, status = PSA_ERROR_NOT_SUPPORTED);
+
+    // Ensure directory
+    VerifyOrExit(ensureDirectoryExists(), status = PSA_ERROR_GENERIC_ERROR);
+
+    // Build path
+    VerifyOrExit(buildFilePath(aUid, path, sizeof(path)) == 0, status = PSA_ERROR_GENERIC_ERROR);
+
+    // If file exists, check WRITE_ONCE
+    file = fopen(path, "rb");
+    if (file)
+    {
+        psa_storage_create_flags_t oldFlags;
+        uint32_t                   oldLen;
+
+        if (readHeader(file, &oldFlags, &oldLen) == 0)
+        {
+            VerifyOrExit(!(oldFlags & PSA_STORAGE_FLAG_WRITE_ONCE), status = PSA_ERROR_NOT_PERMITTED);
+        }
+
+        fclose(file);
+        file = NULL;
+    }
+
+    // Create/overwrite
+    file = fopen(path, "wb");
+    VerifyOrExit(file != NULL, status = PSA_ERROR_GENERIC_ERROR);
+
+    // Write header
+    VerifyOrExit(writeHeader(file, aCreateFlags, aDataLength) == 0, status = PSA_ERROR_GENERIC_ERROR);
+
+    // Write data
+    if (aDataLength > 0 && aData != NULL)
+    {
+        size_t written = fwrite(aData, 1, aDataLength, file);
+        VerifyOrExit(written == aDataLength, status = PSA_ERROR_GENERIC_ERROR);
+    }
+
+exit:
+    if (file)
+    {
+        fclose(file);
```

Comment:
If `fclose` fails, in standard C, it's possible that the file hasn't been fully written. I don't know if that's a concern on OpenThread.